config/aaa.py

```diff
@@ -13,19 +16,27 @@ def is_secret(secret):
 
 
 def add_table_kv(table, entry, key, val):
-    config_db = ConfigDBConnector()
+    config_db = ValidatedConfigDBConnector(ConfigDBConnector())
     config_db.connect()
-    config_db.mod_entry(table, entry, {key:val})
+    try:
+        config_db.mod_entry(table, entry, {key:val})
+    except Exception as e:
```

Comment:
>Exception [](http://example.com/codeflow?start=11&length=9)

I guess you are handling the case that Invalid ConfigDB before you apply patch. If yes, possible to use a more specific exception type?

Let's assume yang model is fancy enough to capture `val` is not invalid, it will also throw and catch here. So the error message may be misleading.

Reply:
Yes, we are handling the case that ConfigDB is invalid before applying patch. But this implementation also works for if yang model catches that `val` is invalid. When we print the exception, as long as the constraint is reflected properly in the yang model, we can get a pretty clear error message. 

For example:
```
admin@vlab-01:~$ sudo config aaa authentication failthrough huh
Patch Applier: Patch application starting.
Patch Applier: Patch: [{"op": "add", "path": "/AAA/authentication/failthrough", "value": "huh"}]
Patch Applier: Getting current config db.
Patch Applier: Simulating the target full config after applying the patch.
Patch Applier: Validating target config does not have empty tables, since they do not show up in ConfigDb.
Patch Applier: Sorting patch updates.
Usage: config aaa authentication failthrough [OPTIONS] OPTION
Try "config aaa authentication failthrough -h" for help.

Error: Invalid ConfigDB. Error: Given patch will produce invalid config. Error: Data Loading Failed
Value "huh" does not satisfy the constraint "false|true|False|True" (range, length, or pattern).
```

We have the option to set specific error-message in YANG models as well. This would require updating the YANG models. I would like to start doing this for scenarios where we have multiple issues throwing the same exception type.

Reply:
Changed to make exception handling more specific